src/components/MakeWidget/subcomponents/RateField/RateField.tsx

```diff
@@ -0,0 +1,29 @@
+import Icon from "../../../Icon/Icon";
+import { Text, Wrapper, RateBox, Button } from "./RateField.styles";
+
+export type RateFieldProps = {
+  TokenPair: string[];
+  Rate: number;
+  onChange: (TokenPair: string[]) => void;
+};
```

Comment:
```suggestion
export type RateFieldProps = {
  token1: string;
  token2: string;
  rate: BigNumber;
  className?: string;
};
```

I think you made the component too simple. Think about someone implementing your component: they just want to import the RateField component, put in 2 token pairs, add a conversion rate and then let the component do the rest. The rate should be BigNumber because potentially the number could be very big and javascript doesn't handle that very well.

Reply:
Makes sense. So it won't be a controlled component then, as all state switching and rate conversion will be handled in the component? The only changes from parent would be passing down a new rate and new pair if the pairing changes, correct?